src/common/types.ts

```diff
@@ -326,3 +326,23 @@ export type FindRouterContextOptions = {
   location: Location;
   basePath?: string;
 };
+
+export type Loader<T = Record<string, any>> = {
+  hydrate?: () => void;
+  beforeLoad?: (params: {
+    prevContext: RouterContext;
+    nextContext: RouterContext;
+  }) => void;
+  load: (context: RouterContext, prevContext?: RouterContext) => T;
+  prefetch: (context: RouterContext) => void;
+};
+
+export type CombinedLoader<T = Record<string, any>> = {
+  hydrate: () => void;
+  beforeLoad: (params: {
```

Comment:
Should this follow the same shape as `load` ? 
```
beforeLoad: (prevContext: RouterContext, nextContext: RouterContext ) => void;
```

Reply:
Yeah, or the other way around is more flexible for future changes.

`beforeLoad({ nextContext, prevContext })`

`Load({ nextContext, prevContext })`